lib/src/license_detection/crc32.dart

```diff
@@ -0,0 +1,291 @@
+// Copyright (c) 2021, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+part of 'license_detector.dart';
+
+/// Computes CRC-32 checksum of [bytes].
+///
+/// This follows the [CRC-32][1] specification used in IEEE 802.3 Ethernet (Normal).
+/// ```
+/// Width  : 32
+/// Poly   : 0x04c11db7
+/// Init   : 0xffffffff
+/// RefIn  : true
+/// RefOut : true
+/// ```
+///
+/// [1]: https://en.wikipedia.org/wiki/Cyclic_redundancy_check
+@visibleForTesting
+int crc32(List<int> bytes) {
+  var crc = 0xffffffff;
+  final length = bytes.length;
+
+  for (var i = 0; i < length; i++) {
+    crc = (crc >> 8) ^ _crc32Table[(crc & 255) ^ bytes[i]];
+  }
+
+  return (~crc).toUnsigned(32);
+}
+
+/// CRC-32 table generated using [Rosetta][1] C implementation code.
+///
+/// [1]: https://rosettacode.org/wiki/CRC-32#Implementation_2
+const _crc32Table = [
+  0,
+  0x77073096,
+  0xee0e612c,
+  0x990951ba,
+  0x76dc419,
+  0x706af48f,
+  0xe963a535,
+  0x9e6495a3,
+  0xedb8832,
+  0x79dcb8a4,
+  0xe0d5e91e,
+  0x97d2d988,
+  0x9b64c2b,
+  0x7eb17cbd,
+  0xe7b82d07,
+  0x90bf1d91,
+  0x1db71064,
+  0x6ab020f2,
+  0xf3b97148,
```

Comment:
To custom format this table you can do something like: 
```suggestion
const _crc32Table = [
  0, 0x77073096, 0xee0e612c, 0x990951ba, 0x76dc419, 0x706af48f, //
  0xe963a535, 0x9e6495a3, 0xedb8832, 0x79dcb8a4, 0xe0d5e91e, 
 ...
```
See https://github.com/dart-lang/dart_style/wiki/FAQ#why-does-the-formatter-mess-up-my-collection-literals

Reply:
Done